### ISSUE TITLE
Anonymous user is able to access unpublished resources

### ISSUE THREAD

**Issue by Pratyush1991** [GeoNode/geonode#7556]
## Expected Behavior
Anonymous users should not be able to access any unpublished resources.

## Actual Behavior
Anonymous users can access an unpublished resource under the user's activity section.

The main page shows no layer for an anonymous user

![image](https://user-images.githubusercontent.com/28738918/118782434-4bcab080-b8b8-11eb-9234-a40f992afe97.png)

A user's activity page can be accessed by an anonymous user and by clicking on the uploaded resource anonymous user can access the resource.

![image](https://user-images.githubusercontent.com/28738918/118782959-ce537000-b8b8-11eb-9d02-3caf38a740f3.png)

![image](https://user-images.githubusercontent.com/28738918/118784343-22128900-b8ba-11eb-90bf-ba439076467c.png)

## Steps to Reproduce the Problem

  1. Set RESOURCE_PUBLISHING=True, GROUP_PRIVATE_RESOURCES=True, Rest all variables as default.
  2. Upload a layer as any registered user then log out.
  3. Check user activity and click on the uploaded layer by the user.

## Specifications

  - GeoNode version: 3.2
  - Installation method: GeoNode Docker
  - Platform: WSL2 Ubuntu 20.04
  - Additional details:

**Comment by giohappy** [comment]
@Pratyush1991 the visibility of the layer is driven by the permissions. How are the permissions for "Anyone" set on your layer?
If the resource is visible to anyone it should appear with the "unpublished" label, see example below.

![image](https://user-images.githubusercontent.com/571129/118800177-e9fe4c80-b89f-11eb-977a-32639473817a.png)

**Comment by Pratyush1991** [comment]
The layer appears as "unpublished" after I log in. If I logout it shows no layer in Layer section. I did not set any layer permission explicitly during upload just uploaded with the default setting. If I'm not wrong the default layer permission was anyone can view, download or change, etc.  I uploaded it as admin.

After I login. 

![image](https://user-images.githubusercontent.com/28738918/118811619-69f3d900-b8d7-11eb-9289-963a2af36e55.png)

After I logout. The Layer Section shows no layer.

![image](https://user-images.githubusercontent.com/28738918/118812224-17ff8300-b8d8-11eb-9e48-9e5c24a97868.png)

But I can access the same layer from the User Activity Page. 

![image](https://user-images.githubusercontent.com/28738918/118812546-6f055800-b8d8-11eb-8d79-0c089b3ab81d.png)
